### PR TITLE
[Fix-4757][worker] fix worker failover host error

### LINKED ISSUE #4757: [Bug][server] failoverWorker 容错
分支：dev
测试中发现dolphinscheduler-server模块中关于ZKMasterClient中failoverWorker方法中查询需要容错的任务实例时，入参workerHost并不是ip地址加端口号
image

根据打印的日志显示如下:
image

从而导致查询不到数据库中的需要容错任务实例，不能完成容错机制。同理failoverMaster也存在这样的问题。

不知是否是我配置问题，还是什么原因。谢谢

### PR DESCRIPTION
**Pull request by zhuangchong**
## *Tips*
- *Thanks very much for contributing to Apache DolphinScheduler.*
- *Please review https://dolphinscheduler.apache.org/en-us/community/index.html before opening a pull request.*

## What is the purpose of the pull request
this PR closes #4757 

fix worker failover host error

## Verify this pull request
  - *Manually verified the change by testing locally.*

image

image

### CODE REVIEW

**Comment by chengshiwen** [comment]
Please pass the checks @zhuangchong

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/WorkerGroupServiceImpl.java [hunk 1]
```diff
@@ -134,45 +136,48 @@
...
             if (e.getMessage().contains(NO_NODE_EXCEPTION_REGEX)) {
-                if (!isPaging) {
+                if (isPaging) {
+                    return workerGroups;
+                } else {
                     //ignore noNodeException return Default
                     WorkerGroup wg = new WorkerGroup();
                     wg.setName(DEFAULT_WORKER_GROUP);
                     workerGroups.add(wg);
+                    return workerGroups;
```
Comment (chengshiwen) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/WorkerGroupServiceImpl.java]: If there is no better modification, it's better to keep the old code
image

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/zk/ZookeeperNodeHandler.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    /**
+     * get worker startTime
+     * @param workerZkNode worker zookeeper node
+     * @return worker startTime
+     */
+    public static String getWorkerStartTime(WorkerZkNode workerZkNode) {
+        return workerZkNode == null ? null : workerZkNode.getStartTime();
+    }
+
+}
```
Comment (chengshiwen) [inline dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/zk/ZookeeperNodeHandler.java]: Require zookeeper unit test. Now the unit test coverage is less than 33%

**Review by chengshiwen — CHANGES_REQUESTED** [review]

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/model/WorkerZkNode.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dolphinscheduler.common.model;
+
+/**
+ * worker zookeeper node
+ */
+public class WorkerZkNode {
```
Comment (CalvinKirs) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/model/WorkerZkNode.java]: We have calculations similar to this in many places(eg: Host. class), can we organize them into one?

Comment (zhuangchong) [reply dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/model/WorkerZkNode.java]: done.

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Pull request merged**